pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -283,40 +271,36 @@ int sock_dtls_session_create(sock_dtls_t *sock, const sock_udp_ep_t *ep,
     memcpy(&remote->dtls_session.addr, &ep->addr.ipv6, sizeof(ipv6_addr_t));
     _ep_to_session(ep, &remote->dtls_session);
 
-    /* start a handshake */
-    DEBUG("sock_dtls: starting handshake\n");
-    res = dtls_connect(sock->dtls_ctx, &remote->dtls_session);
+    sock_udp_ep_t local;
+    if (sock_udp_get_local(sock->udp_sock, &local) < 0) {
```

Comment:
Can `sock->udp_sock` be `NULL` here?

Reply:
No, I don't think so. `sock_udp_get_local` checks `local` not `NULL`

Reply:
No, it *asserts* `sock != NULL` and `local != NULL`, so if `sock->udp_sock` is `NULL` a program might crash. I don't know enough to know under which condition `sock->udp_sock` is not NULL. If it is _only_ when `sock` is not bound, this should be checked! Alternatively, we can remove the `-EADDRNOTAVAIL` case and also assert, that `sock` is bound (i.e. for `tinydtls` `assert(sock->udp_sock)` when my previous statement was right).

Reply:
`sock_dtls_create` which should be called first before calling `sock_dtls_session_init` already asserts that `udp_sock != NULL`. For APIs generally, do we also need to check if the user is calling the function in the wrong order?

For tinydtls, we don't require that the `udp_sock` is bound to a local `sock_udp_ep_t`. We specify the endpoint when calling `sock_udp_send` (see `_write`). 

> Alternatively, we can remove the `-EADDRNOTAVAIL` case and also assert, that `sock` is bound (i.e. for `tinydtls` `assert(sock->udp_sock)` when my previous statement was right).

So I think we should do this.

Reply:
I prefer the soft solution in this case (check if `sock->udp_sock` is set). Calling order should not be expected. There are enough people out there who don't follow what they (expectedly) read and then say things are broken ;-)

Reply:
(also this is more preferable since it is not as hard an API change compared to the current behavior)

Reply:
So would this be okay?
```
if (!sock->udp_sock || (sock_udp_get_local(sock->udp_sock, &local) < 0)) {
        return -EADDRNOTAVAIL;
}
```

Reply:
Yepp :-).